### PR TITLE
[Merged by Bors] - feat: port AlgebraicTopology.DoldKan.EquivalenceAdditive

### PR DESCRIPTION
**Pull request by joelriou**
---
- [x] depends on: #3566
- [x] depends on: #3568
- [x] depends on: #3576

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3566
* leanprover-community/mathlib4#3568
* leanprover-community/mathlib4#3576
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3566~~
* ~~leanprover-community/mathlib4#3568~~
* ~~leanprover-community/mathlib4#3576~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
Merged master so github shows the correct diff.

File: Mathlib/AlgebraicTopology/DoldKan/EquivalenceAdditive.lean [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  counitIso := N₂Γ₂
+  functor_unitIso_comp P := by
+    -- porting note: the proof had to be tweaked to avoid timeouts
+    suffices N₂Γ₂.inv.app (N₂.obj P) = N₂.map (Γ₂N₂.hom.app P) by
+      dsimp only [N]
+      erw [← this, ← N₂Γ₂.inv_hom_id_app (N₂.obj P)]
+    rw [← cancel_mono (N₂.map (Γ₂N₂.natTrans.app P)),
+      AlgebraicTopology.DoldKan.identity_N₂_objectwise P, ← N₂.map_comp]
+    simp only [← Γ₂N₂_inv, ← NatTrans.comp_app, Iso.hom_inv_id, NatTrans.id_app,
+      Functor.id_obj, N₂.map_id]
```
Comment (semorrison) [inline Mathlib/AlgebraicTopology/DoldKan/EquivalenceAdditive.lean:67]:
```suggestion
    let α := N.mapIso (Γ₂N₂.app P)
    let β := N₂Γ₂.app (N.obj P)
    symm
    change 𝟙 _ = α.hom ≫ β.hom
    rw [← Iso.inv_comp_eq, comp_id, ← comp_id β.hom, ← Iso.inv_comp_eq]
    exact AlgebraicTopology.DoldKan.identity_N₂_objectwise P
```
The original proof works fine for me.

**Comment by semorrison** [comment]
bors d+

**Comment by joelriou** [comment]
Thanks very much @semorrison for the reviews!

bors r+

**Pull request closed without merging**